test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java

```diff
@@ -0,0 +1,393 @@
+        Configuration config = configAdmin.getConfiguration(PolicyProperties.FACTORY_PID, null);
+        Dictionary<String, ?> configProps = new Hashtable<>(policyProperties);
+        config.update(configProps);
+        waitForAllBundles();
+    }
+
+    private void configureRestForBasic() throws IOException, InterruptedException {
+        PolicyProperties policyProperties = new PolicyProperties();
+        policyProperties.put("authenticationTypes", "/=SAML|basic,/admin=SAML|basic,/jolokia=SAML|basic,/system=SAML|basic,/solr=SAML|PKI|basic");
+        policyProperties.put("whiteListContexts", "/services/SecurityTokenService,/services/internal,/proxy," + SERVICE_ROOT + "/sdk/SoapService");
+        Configuration config = configAdmin.getConfiguration(PolicyProperties.FACTORY_PID, null);
+        Dictionary<String, ?> configProps = new Hashtable<>(policyProperties);
+        config.update(configProps);
+        waitForAllBundles();
+    }
+
+    @Test
+    public void testAnonymousSoapAccess() throws Exception {
+        configureRestForAnonymous();
+
+        String body = "<soapenv:Envelope xmlns:hel=\"http://ddf.sdk/soap/hello\" xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                + "   <soapenv:Header>\n" + "   </soapenv:Header>\n" + "   <soapenv:Body>\n" + "      <hel:helloWorld/>\n" + "   </soapenv:Body>\n"
+                + "</soapenv:Envelope>";
+        //we are only testing anonymous because that hits the most code, testing with an assertion would be mostly testing the same stuff that this is hitting
+        given().log().all().body(body).header("Content-Type", "text/xml; charset=utf-8").header("SOAPAction", "helloWorld").expect().statusCode(200)
+                .when().post(SERVICE_ROOT + "/sdk/SoapService").then().log().all()
+                .assertThat().body(HasXPath.hasXPath("//*[local-name()='helloWorldResponse']/result/text()"));
+    }
+
+    /* These STS tests are here to prove out functionality that doesn't get hit when accessing internal services. The standard UsernameToken and BinarySecurityToken elements are supported
+     * by DDF, but not used internally. These elements need to be checked for functionality independently since going through our REST security framework won't touch these validators. */
+
+    @Test
+    public void testUsernameTokenSTS() throws Exception {
+        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'.500Z'");
+        format.setCalendar(calendar);
+        String created = format.format(new Date(calendar.getTimeInMillis()));
+        long now = calendar.getTimeInMillis();
+        now += 60000;
+        String expires = format.format(new Date(now));
+        String body = "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
```

Comment:
Yeah, I considered doing that or dropping them into a file and reading them out. I could go either way, but I leaned towards leaving them here because it actually helped me while I was creating these. It was beneficial to see the request that I was trying to send without going hunting for it in another file. If people have heartburn about leaving them in here, then I'll move them. Otherwise, they will stay here.